matrix-sdk/src/main/java/org/matrix/androidsdk/login/RegistrationTools.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.androidsdk.login
+
+import org.matrix.androidsdk.rest.client.LoginRestClient
+import org.matrix.androidsdk.rest.model.login.LocalizedFlowDataLoginTerms
+import org.matrix.androidsdk.rest.model.login.RegistrationFlowResponse
+
+/**
+ * Get the public key for captcha registration
+ *
+ * @return public key
+ */
+fun getCaptchaPublicKey(registrationFlowResponse: RegistrationFlowResponse?): String? {
+    var publicKey: String? = null
+    registrationFlowResponse?.params?.let {
+        val recaptchaParamsAsObject = it.get(LoginRestClient.LOGIN_FLOW_TYPE_RECAPTCHA)
+        if (recaptchaParamsAsObject is Map<*, *>) {
+            publicKey = recaptchaParamsAsObject["public_key"] as String
+        }
+    }
+    return publicKey
+}
+
+private data class UrlAndName(val url: String, val name: String)
+
+/**
+ * This method extract the policies from the login terms parameter, regarding the user language.
+ * For each policy, if user language is not found, the default language is used and if not found, the first url and name are used (not predictable)
+ *
+ * Example of Data:
+ * <pre>
+ * "m.login.terms": {
+ *       "policies": {
+ *         "privacy_policy": {
+ *           "version": "1.0",
+ *           "en": {
+ *             "url": "http:\/\/matrix.org\/_matrix\/consent?v=1.0",
+ *             "name": "Terms and Conditions"
+ *           }
+ *         }
+ *       }
+ *     }
+ *</pre>
+ *
+ * @param registrationFlowResponse the registration flow to extract data from
+ * @param userLanguage the user language
+ * @param defaultLanguage the default language to use if the user language is not found for a policy in registrationFlowResponse
+ */
+fun getLocalizedLoginTerms(registrationFlowResponse: RegistrationFlowResponse?,
```

Comment:
This method is a bit hard to read...

Reply:
yes, it is because of the model... And also there are unitary tests to cover this function